flink-kubernetes/src/main/java/org/apache/flink/kubernetes/highavailability/KubernetesLeaderElectionDriver.java

```diff
@@ -235,9 +238,16 @@ public void onError(List<KubernetesConfigMap> configMaps) {
 
         @Override
         public void handleFatalError(Throwable throwable) {
-            fatalErrorHandler.onFatalError(
-                    new LeaderElectionException(
-                            "Error while watching the ConfigMap " + configMapName, throwable));
+            if (throwable instanceof KubernetesTooOldResourceVersionException) {
+                LOG.info("Creating a new watch on ConfigMap {}.", configMapName);
+                kubernetesWatch =
+                        kubeClient.watchConfigMaps(
+                                configMapName, new ConfigMapCallbackHandlerImpl());
```

Comment:
I think this violates the threading model of this component which does not allow state changes from a different thread. We either need to make this component thread safe or tell the `LeaderElectionEventHandler` about the required restart.